### PR TITLE
org.signal.Signal

### PR DESCRIPTION
**Pull request by Barthalion**

### CODE REVIEW

**Comment by Barthalion** [comment]
bot, build org.signal.Signal

**Comment by Barthalion** [comment]
The alternative is trying to do that from source and being disappeared like manuq was. Until we have at least partially universal solution, I don't see it as a problem.

**Comment by nedrichards** [comment]
I'm fine with it for now.

File: org.signal.Signal.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  </description>
+  <screenshots>
+    <screenshot>
+      <image type="source">https://signal.org/blog/images/signal-desktop-splash.png</image>
+    </screenshot>
+    <screenshot>
+      <image type="source">https://signal.org/blog/images/signal-desktop-foucault.png</image>
+    </screenshot>
+  </screenshots>
+  <!-- These are the icons for the android apps from Google Play -->
```
Comment (TingPing) [inline org.signal.Signal.appdata.xml]: Adding `<releases>` would be nice.

File: org.signal.Signal.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "build-commands": [
+                "install apply_extra /app/bin",
+                "install signal.sh /app/bin/signal",
+                "install -Dm644 org.signal.Signal.appdata.xml /app/share/appdata/org.signal.Signal.appdata.xml",
+                "cp /usr/bin/ar /app/bin",
+                "cp /usr/lib/libbfd-*.so /app/lib"
+            ],
+            "sources": [
+                {
+                    "type": "extra-data",
```
Comment (TingPing) [inline org.signal.Signal.json]: The project is GPL3, we can redistribute it directly which avoids some problems and simplifies the manifest.

Comment (Barthalion) [reply org.signal.Signal.json]: Fair point. I will change it tomorrow.

File: org.signal.Signal.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "command": "signal",
+    "separate-locales": false,
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--share=network",
+        "--device=dri",
+        "--filesystem=home",
+        "--talk-name=org.freedesktop.Notifications"
```
Comment (TingPing) [inline org.signal.Signal.json]: Does it support tray icons? If so you'll need the env var and talk names from Discord for example.

Comment (Barthalion) [reply org.signal.Signal.json]: It doesn't, at least in this version. (There is pull request that adds it but not merged yet.)

**Comment by manuq** [comment]
@Barthalion I haven't dissapeared :) just busy. But yes long term I would like to enhace flatpak to have a new source type "yarn" or "npm" that would take a lockfile and bring up all the dependencies. For now it seems easier to use a released binary with extra-data.

File: org.signal.Signal.appdata.xml (hunk 1 above)
Comment (Barthalion) [reply org.signal.Signal.appdata.xml]: Done.

**Comment by Barthalion** [comment]
bot, build

**Comment by Barthalion** [comment]
bot, build org.signal.Signal

**Comment by Barthalion** [comment]
One that worked: https://flathub.org/builds/#/builders/1/builds/850

**Pull request closed without merging**